### ISSUE TITLE
Regarding model training questions

### ISSUE THREAD

**Issue by pengjuRen99** [AzureLeon1/DiffSpectra#2]
Hello, this is very meaningful work, and I'm very interested! I have some questions for you. 
During model training, is the training set not divided into two parts, but rather the entire training set is used for training? I noticed that previous works divide the conditional generation into two parts (first_train_dataset, second_train_dataset), but in your recently uploaded code, you modified line 45 of build_dataset.py to second_train_dataset = dataset.index_select(train_idx).
I'd like to know if, if I pre-train the specformer, can I only pre-train on first_train_dataset and train the generation task on second_train_dataset? If the entire training set is used for specformer pre-training, is there a problem with information leakage? Can the weights of the pre-trained Specformer be provided? (I seem to have forgotten to upload this.) I look forward to your reply.

**Comment by AzureLeon1** [comment]
Thanks for your interest in our work!

First, some prior conditional molecular generation studies (e.g., property-based molecular generation models) divide the training set into two parts because they need to evaluate whether their generated molecules match the desired properties. To do this, they use part of the training data to train a property-prediction model for evaluation.

In our molecular structure elucidation setting, such a division is unnecessary because we can introduce more comprehensive metrics (e.g., top-k accuracy, similarity scores, RMSD) to evaluate the generated molecular structures. Therefore, a natural choice in our setting is to use the entire training set to train our generative model. These data can also be used to pre-train SpecFormer without causing any information-leakage issues, since the test data in the elucidation task are not used for pre-training SpecFormer or for training DiffSpectra.

And thank you for the reminder. I'll be releasing our pre-trained SpecFormer on HuggingFace soon.

**Comment by pengjuRen99** [comment]
Thank you very much for your timely reply. I have understood now.
